### PR TITLE
[#4181]Improvement: Support setting gradle property skipDockerTests by environment variant

### PR DESCRIPTION
**Pull request by featherchen**
### What changes were proposed in this pull request?

Read the environment variable (if exists) when -PskipDockerTests is not specified.

### Why are the changes needed?

Fix: #4181 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

```
export SKIP_DOCKER_TESTS=false
./gradlew build
# check docket tests are running
```

### CODE REVIEW

**Comment by yuqi1129** [comment]
Can you please add some description in the doc https://github.com/apache/gravitino/blob/80e1fc72a0e0ee5aa243d897f368b0f17b058482/docs/how-to-build.md?plain=1#L28-L30

**Comment by featherchen** [comment]
> Can you please add some description in the doc
> 
> https://github.com/apache/gravitino/blob/80e1fc72a0e0ee5aa243d897f368b0f17b058482/docs/how-to-build.md?plain=1#L28-L30

Sure, I have updated the doc.

**Comment by yuqi1129** [comment]
LTGM
@mchades  would you like to take a look?

**Comment by xunliu** [comment]
Do we need to keep `skipDockerTests` properties in the https://github.com/apache/gravitino/blob/main/gradle.properties#L43 ?

**Comment by yuqi1129** [comment]
> @yuqi1129 @mchades @featherchen Do we need to keep `skipDockerTests` propertie in the [`main`/gradle.properties#L43](https://github.com/apache/gravitino/blob/main/gradle.properties?rgh-link-date=2024-07-22T10%3A51%3A51Z#L43) ?

If we do not add it in the `gradle.properties`, what's the default value of `skipDockerTests`?

File: build.gradle.kts [hunk 1]
```diff
@@ -690,8 +690,12 @@ fun printDockerCheckInfo() {
   val dockerRunning = project.extra["dockerRunning"] as? Boolean ?: false
   val macDockerConnector = project.extra["macDockerConnector"] as? Boolean ?: false
   val isOrbStack = project.extra["isOrbStack"] as? Boolean ?: false
-
-  if (extra["skipDockerTests"].toString().toBoolean()) {
+  val skipDockerTest = if (extra["skipDockerTests"].toString().toBoolean()) {
+    System.getenv("SKIP_DOCKER_TESTS")?.toBoolean() ?: true
+  } else {
+    false
+  }
```
Comment (mchades) [inline build.gradle.kts:697]: Please add comments to clarify the logic.

Comment (featherchen) [reply build.gradle.kts]: I have add some comments, hope it could help to make the logic clearer.

**Comment by featherchen** [comment]
> > @yuqi1129 @mchades @featherchen Do we need to keep `skipDockerTests` propertie in the [`main`/gradle.properties#L43](https://github.com/apache/gravitino/blob/main/gradle.properties?rgh-link-date=2024-07-22T10%3A51%3A51Z#L43) ?
> 
> If we do not add it in the `gradle.properties`, what's the default value of `skipDockerTests`?

Furthermore, if we do not add it, we still need to give the default value of it in related parts of `build.gradle.kts`

**Review by mchades — APPROVED** [review]
LGTM, do you have more comments? @xunliu @yuqi1129

**Comment by yuqi1129** [comment]
> LGTM, do you have more comments? @xunliu @yuqi1129

I'm okay with the current changes.

**Pull request merged**